### PR TITLE
feat: add a unique user agent

### PR DESCRIPTION
**Pull request by gssbzn**
## Description

We are trying to track usage of the API and right now is hard to distinguish from the TF provider and a vanilla go client usage.

I'm adding a user agent here to help us better track usage.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the MongoDB CLA
- [x] I have read the Terraform contribution guidelines 
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirments
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: mongodbatlas/config.go
Comment (MihaiBojin) [inline mongodbatlas/config.go:30]: it would be great if we could add the provider's version too!

Comment (gssbzn) [reply mongodbatlas/config.go:30]: the provider looks to be unversioned (no ldflags), I'll leave the maintainer to decide if we want to add this (in this PR or a follow up) or not

**Comment by gssbzn** [comment]
Linting seems to be failing on travis, I had plans to fix lint installation later but Opening a PR now

**Comment by gssbzn** [comment]
Opened #269 to fix linting

**Review by PacoDw — APPROVED** [review]
LGMT! Thanks

File: mongodbatlas/config.go
Comment (MihaiBojin) [reply mongodbatlas/config.go:30]: Updating the UA like so obscures the underlying go client's version.
I think `atlasClient.SetUserAgent(...)` is a better approach, but it's not a hard requirement from my point of view, since we'll version the TF provider soon and we'll be able to infer which go-client it's using.

**Review by MihaiBojin — APPROVED** [review]

File: mongodbatlas/config.go
Comment (themantissa) [reply mongodbatlas/config.go:30]: Ah, so this is kinda a weirdness of Terraform and how we ended up not tracking the version before.  Per https://www.terraform.io/docs/extend/how-terraform-works.html#plugin-locations "Provider and provisioner plugins can be installed in the same directories. Provider plugin binaries are named with the scheme terraform-provider-<NAME>_vX.Y.Z, while provisioner plugins use the scheme terraform-provisioner-<NAME>_vX.Y.Z. **Terraform relies on filenames to determine plugin types, names, and versions.**". So they are versioned by filename (I know).   I don't think there would be an issue versioning with ldflags (but leave it to the DoU team to chime in on this) once we move to our repo again and our own releases.  

So that was my long way of saying we can solve this post repo move (hopefully).

**Review by themantissa — APPROVED** [review]
LGTM!  Making note to solve versioning within the code base for tracking for later.

**Pull request merged**
